src/host/ft_host/API_OutputTests.cpp

```diff
@@ -363,12 +364,6 @@ void OutputTests::WriteConsoleOutputWNegativePositions()
             VERIFY_ARE_EQUAL(expectedItem, readItem);
         }
     }
-
-    // Set the region so the left will end up past the right
-    adjustedRegion = region;
-    adjustedRegion.Left = -(adjustedRegion.Right + 1);
-    affected = adjustedRegion;
-    VERIFY_WIN32_BOOL_FAILED(WriteConsoleOutputW(consoleOutputHandle, buffer.data(), regionDimensions, regionOrigin, &affected));

```

Comment:
we don't want the API to fail in this case?

Reply:
Yes, it should succeed, it makes absolutely no sense otherwise. From what I can tell by reading just the source code, this was a regression introduced in Windows Vista, most likely as a crash fix for conhost.

However, the person who wrote the crash fix didn't seem to understand how the code worked.

In Windows 2000, where this code was first added, we had something like this (pseudo-code, comments are mine):
```cpp
NTSTATUS WriteScreenBuffer(...) {
    // Source buffer empty?
    if (SourceSize.X <= 0 || SourceSize.Y <= 0) {
        return STATUS_SUCCESS;
    }
    // Target area is outside the buffer...
    // ...but it only checks whether it's too far right / below??
    if (psrWriteRegion->Left >= BufferSize.X || psrWriteRegion->Top >= BufferSize.Y) {
        return STATUS_SUCCESS;
    }

    // ... clamping ...

    return WriteRectToScreenBuffer(...);
}
```

Then in Windows Vista it looks like this:
```cpp
NTSTATUS WriteScreenBuffer(...) {
    // Source buffer empty?
    if (SourceSize.X <= 0 || SourceSize.Y <= 0
    ) {
        return STATUS_SUCCESS;
    }
    // Target area is outside the buffer...
    // ...but it only checks whether it's too far right / below??
    if (psrWriteRegion->Left >= BufferSize.X || psrWriteRegion->Top >= BufferSize.Y) {
        return STATUS_SUCCESS;
    }

    // ... clamping ...

    // Clamped rect is empty...
    // ...but the check is completely wrong??
    // The clamping logic assumes that the target rect is in bounds!
    if (clamped rect is empty) {
        return STATUS_INVALID_PARAMETER;
    }

    return WriteRectToScreenBuffer(...);
}
```

What the person who changed this code failed to understand is that the issue isn't a missing clamp, the issue is that the 2nd if condition is just plain wrong. The clamping logic then relies on the incorrect assumption that the if condition is correct which results in a completely incorrect clamped region (the clamped region can be outside the buffer to the left or above with negative coordinates among others).

In addition to the 2nd if condition returning `STATUS_SUCCESS` for targets that are too far to the right/below, [MSDN also clearly says](https://learn.microsoft.com/en-us/windows/console/writeconsoleoutput):
> If the rectangle specified by `lpWriteRegion` lies completely outside the boundaries of the console screen buffer, [...] no data is written. In this case, the function returns [...] `lpWriteRegion` parameter set such that [it's empty]

...implying that it should also return `STATUS_SUCCESS` for targets that are too far to the left or top.